src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -256,3 +264,40 @@ doLookupSuccessful dbenv (TypedHash hash) = runBlockEnv dbenv $ do
         !hsh <- either fail return $ Data.Serialize.decode blob
         return $! T2 (fromIntegral h) hsh
     go _ = fail "impossible"
+
+doGetBlockHistory :: FromJSON v => Db -> BlockHeader -> Domain k v -> IO BlockTxHistory
+doGetBlockHistory dbenv blockHeader d = runBlockEnv dbenv $ do
+  callDb "doGetBlockHistory" $ \db -> do
+    endTxId <- getEndTxId db bHeight (_blockHash blockHeader)
+    startTxId <- getEndTxId db (pred bHeight) (_blockParent blockHeader)
+    history <- queryHistory db (domainTableName d) startTxId endTxId
+    return $! BlockTxHistory $ V.fromList $ M.toList $ foldl' groupByTxid mempty history
+  where
+
+    bHeight = _blockHeight blockHeader
+
+    groupByTxid :: Ord a => M.Map a [b] -> (a,b) -> M.Map a [b]
+    groupByTxid r (t,l) = M.insertWith (++) t [l] r
+
+    getEndTxId db bhi bha = do
+      r <- qry db
+        "SELECT endingtxid FROM BlockHistory WHERE blockheight = ? and hash = ?;"
+        [SInt $ fromIntegral $ bhi, SBlob $ encode $ bha]
+        [RInt]
+      case r of
+        [[SInt tid]] -> return tid
+        [] -> throwM $ BlockHeaderLookupFailure $ "doGetBlockHistory: not in db: " <>
+              sshow (bhi,bha)
+        _ -> internalError $ "doGetBlockHistory: expected single-row int result, got " <> sshow r
+
+    queryHistory db tableName s e = do
```

Comment:
can you add type information for `queryHistory`? and a comment on whether the start and end index are inclusive?